src/Symfony/Component/HttpClient/Retry/RetryDeciderInterface.php

```diff
@@ -11,15 +11,17 @@
 
 namespace Symfony\Component\HttpClient\Retry;
 
-use Symfony\Contracts\HttpClient\ResponseInterface;
-
 /**
  * @author Jérémy Derussé <jeremy@derusse.com>
  */
 interface RetryDeciderInterface
 {
     /**
      * Returns whether the request should be retried.
+     *
+     * @param ?string $responseContent contains null when the body is not yet fully arrived
+     *
+     * @return ?bool returns null when the service needs body to take a decision
      */
-    public function shouldRetry(string $requestMethod, string $requestUrl, array $requestOptions, ResponseInterface $partialResponse, \Throwable $throwable = null): bool;
+    public function shouldRetry(string $requestMethod, string $requestUrl, array $requestOptions, int $responseStatusCode, array $responseHeaders, ?string $responseContent, \Throwable $throwable = null): ?bool;
```

Comment:
we could merge both last arguments because it should be impossible to have both the content and an exception at hand
`string|\Throwable|null $responseContentOrError`
For thoughts, I'm not sure that's what we should do :)

Reply:
BTW, TransportExceptionInterface instead of Throwable?

Reply:
I'm not sure about property mixing different type and context.

BTW, does it make sens to pass the exception to the strategy shouldn't we always (of course, following the maxRetry param) retry a NetworkException?

Reply:
I agree, let's always retry on exception.

Reply:
Done. Sorry, the diff is a less readable.